sequence-utils.dylan

```diff
@@ -588,3 +588,29 @@ define method precedes?(elt-1, elt-2, seq :: <sequence>,
     not-found;
   end block;
 end method precedes?;
+
+define function zip
```

Comment:
Naming is hard, so I won't claim that this is definitely better, but one potential alternative is to call this function `zip2` and call `zip-all` just `zip` so that it matches other popular languages (Python).